gengokit/template/NAME-service/NAME-cli-client/client_main.gotemplate

```diff
@@ -31,27 +31,47 @@ var (
 
 
 func main() {
+	var (
+		httpAddr       = flag.String("http.addr", "", "HTTP address of addsvc")
+		grpcAddr       = flag.String("grpc.addr", ":5040", "gRPC (HTTP) address of addsvc")
+	)
+
+
 	// The addcli presumes no service discovery system, and expects users to
 	// provide the direct address of an service. This presumption is reflected in
 	// the cli binary and the the client packages: the -transport.addr flags
 	// and various client constructors both expect host:port strings.
 
-	var (
-		httpAddr       = flag.String("http.addr", "", "HTTP address of addsvc")
-		grpcAddr       = flag.String("grpc.addr", ":5040", "gRPC (HTTP) address of addsvc")
-		method         = flag.String("method", "{{range $index, $i := .Service.Methods}}{{if $index}}{{else}}{{ToLower $i.Name}}{{end}}{{end}}",	"{{range $index, $i := .Service.Methods}}{{if $index}},{{end}}{{ToLower $i.Name}}{{end}}")
-	)
+	{{ range $meth, $args := .ClientArgs.MethArgs }}
+	fs{{$meth}} := flag.NewFlagSet("{{ToLower $meth}}", flag.ExitOnError)
+	{{end}}
 
 	var (
 		{{.ClientArgs.AllFlags}}
 	)
-	flag.Parse()
 
+	flag.Usage = func() {
+		fmt.Fprintf(os.Stderr, "Usage of %s:\n", os.Args[0])
+		flag.PrintDefaults()
+		fmt.Fprintf(os.Stderr, "Subcommands:\n")
```

Comment:
Nit, perhaps add a newline here (i.e. `"\nSubcommands:\n"`) as I think the output would be nicer.

Current:
```
$ echo-cli-client
Usage of echo-cli-client:
  -grpc.addr string
        gRPC (HTTP) address of addsvc (default ":5040")
  -http.addr string
        HTTP address of addsvc
Subcommands:
  echo
  louder
$
```
Newline:
```
$ echo-cli-client
Usage of echo-cli-client:
  -grpc.addr string
        gRPC (HTTP) address of addsvc (default ":5040")
  -http.addr string
        HTTP address of addsvc

Subcommands:
  echo
  louder
$ 
```